src/GitHub.Exports/Services/GitCloneCommandLineService.cs

```diff
@@ -0,0 +1,139 @@
+        {
+            this.vsAppCommandLine = vsAppCommandLine;
+            this.vsGitServices = vsGitServices;
+            this.vsServices = vsServices;
+            this.operatingSystem = operatingSystem;
+        }
+
+        public string FindGitCloneOption()
+        {
+
+            if (vsAppCommandLine == null)
+            {
+                return null;
+            }
+
+            int isPresent;
+            string optionValue;
+            if (ErrorHandler.Failed(vsAppCommandLine.GetOption("GitClone", out isPresent, out optionValue)))
+            {
+                return null;
+            }
+
+            if (isPresent == 0)
+            {
+                return null;
+            }
+
+            return optionValue;
+        }
+
+        public bool TryOpenRepository(string cloneUrl)
+        {
+            if(TryOpenLocalClonePath(cloneUrl))
+            {
+                return true;
+            }
+
+            if(TryOpenKnownRepository(cloneUrl))
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        bool TryOpenKnownRepository(string cloneUrl)
+        {
+            var repos = vsGitServices.GetKnownRepositories();
+            foreach (var repo in repos)
+            {
+                if (cloneUrl == repo.CloneUrl)
+                {
+                    if(vsServices.TryOpenRepository(repo.LocalPath))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        bool TryOpenLocalClonePath(string cloneUrl)
+        {
+            var cloneUri = FindGitHubCloneUri(cloneUrl);
+            if(cloneUri == null)
+            {
+                return false;
+            }
+
+            var clonePath = vsGitServices.GetLocalClonePathFromGitProvider();
+            if(clonePath == null)
+            {
+                return false;
+            }
+
+            var repoPath = Path.Combine(clonePath, cloneUri.Owner, cloneUri.RepositoryName);
+            var repoDir = operatingSystem.Directory.GetDirectory(repoPath);
+            if (!repoDir.Exists)
+            {
+                return false;
+            }
+
+            return vsServices.TryOpenRepository(repoPath);
+        }
+
+        static UriString FindGitHubCloneUri(string cloneUrl)
+        {
+            try
+            {
+                var uriString = new UriString(cloneUrl);
+                if (uriString.Host != "github.com")
+                {
+                    return null;
+                }
+
+                if(string.IsNullOrEmpty(uriString.Owner) || string.IsNullOrEmpty(uriString.RepositoryName))
+                {
+                    return null;
+                }
+
+                return uriString;
+            }
+            catch(Exception e)
+            {
+                Trace.WriteLine(e);
```

Comment:
The `VsOutputLogger` class provides logging services for code running at startup and in Team Explorer. It doesn't do Trace logging, but there's a trace listener there if you want to hook it up (or call Trace from the methods there if you want to do it the other way around).

For other logging, we're using NLog right now, btw. PR #573 introduces Serilog for logging purposes, which is a much lighter weight solution for logging that we could use for everything. Maybe that's something worth revisiting and cleaning up so we can fix up our logging once and for all.